controllers/apps/cluster_controller.go

```diff
@@ -131,6 +131,8 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 			&clusterHaltTransformer{},
 			// handle cluster deletion
 			&clusterDeletionTransformer{},
+			// handle cluster pause and resume
+			&clusterPauseTransformer{},
```

Comment:
Should the pause and resume operations be executed before all transformers? According to your design, can a cluster that is being deleted be paused?

Reply:
Deletion has a higher priority than pause in my design, which is refer to the design of rollout pause of the k8s deployment.